### DISCUSSION TITLE
Forms: rules() to accept a closure

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#1911] [Ideas]
Just come across a use case where my options for a select are dynamically generated based upon the input of another field. In this instance, my validation would change.

```
Select::make('type')
   ->options([
      'typeone' => 'Type One',
      'typetwo' => 'Type Two'
   ->reactive(),
Select::make('specific')
    ->options(function($get) {
        return match($get('type')) {
            'typeone' => ['specificone' => 'Specific One', 'specifictwo' => 'Specific Two'],
            'typetwo' => ['specificthree' => 'Specific Three', 'specificfour' => 'Specific Four']
        }
    })
    ->rules( //cannot set dynamically );
```

Currently, `rules()` only accepts a string or an array. Would be great if this could accept a closure also to enable dynamic validation like this.

Again, adding here for a reminder. Happy to generate a PR if no objections found. Should be pretty straightforward I believe?

**Comment by danharrin** [comment]
We do support closures, you can pass an array of closures and each closure returns a dynamic rule.

**Reply by titantwentyone** [reply]
Thanks Dan. I didn't see this - it is actually documented [here](https://filamentphp.com/docs/2.x/forms/validation#custom-rules). Apologies!

**Reply by danharrin** [reply]
You can also add "conditional" rules - `->rule(rule, fn ($get, $set, $livewire, $state) => return true if this rule should be applied)`
